api/console_message.go

```diff
@@ -0,0 +1,9 @@
+package api
+
+// ConsoleMessage represents a page console message.
+type ConsoleMessage interface {
```

Comment:
I knew this was gonna be a comment, I had doubts myself :smile: 
Because we've always had this idea of possibly providing other browser implementations I thought that it was better to stick with the current pattern of working with interfaces defined in the API until we take a final decision. But I'm OK moving this to a struct.

Reply:
I have changed `ConsoleMessage` to be defined as a struct in API pkg here: [21c7a5d](https://github.com/grafana/xk6-browser/pull/1006/commits/21c7a5d055f8c11a732a291c0d641efde118fa91). I have also amended the posterior commits to adapt to this change. The only relevant modification happens in [94fbd6c](https://github.com/grafana/xk6-browser/pull/1006/commits/94fbd6c4fcada8568cce9b1914d0869a1fef99ff) as we have to define "allowed redundant mappings" in the mapping layer tests, as `page()`, `text()` and `type()` methods have been defined in the mapping layer explicitly in order to comply with Playwright's API.
cc: @inancgumus

Reply:
I think these methods are not redundant and should be tested if they conform to an API. Since there is not such an API in the `api` package, we can solidify its protocol in tests instead. Like this:

```
type consoleMessageAPI interface {
	Args() []api.JSHandle
	Page() api.Page
	Text() string
	Type() string
}
```

Then in the test:

```
"mapConsoleMessage": {
	apiInterface: (*consoleMessageAPI)(nil), // <---
	mapp: func() mapping {
		return mapConsoleMessage(moduleVU{VU: vu}, &api.ConsoleMessage{})
	},
},
```

I believe this is also a good step in making progress for issue #898, and also provides a solution for my earlier question: "How the testing of the mapping layer is affected?".

Reply:
> Since no such an API exists in the api package, we can solidify the ConsoleMessage protocol in tests instead.

Changed in [3eff159](https://github.com/grafana/xk6-browser/pull/1006/commits/3eff159e72b1609ca8e117057f3fcab8b48075a3). I set this as an anonymous function by now, as this is the only occurrence. If we ever remove the whole API pkg we can define the set of interfaces separately just for the mapping layer testing.

Reply:
Perfect!

Reply:
Perfect! The only downside of an _anonymous interface_ is we'll get no type information when the mapping tests fail. I think we can declare a new type for `ConsoleMessage` in the same test to prevent this.

Reply:
> I realized now that the only downside of an anonymous interface is we'll get no type information when the mapping tests fail

I found it strange that the subtest name was not referenced on failures, so fixed small issues in #1014. With that there is also no need to reference the type name in the error messages, as it is already identified through the subtest name.

Reply:
Thanks for fixing this. If we had declared a named type, this wouldn't be needed. It's okay this way, too, but I wonder what would happen if we wanted to add custom mappings for the anonymous type? A named type makes more sense.

Reply:
> If we had declared a named type, this wouldn't be needed, right? It's okay this way, too, but I wonder what would happen if we wanted to add custom mappings for the anonymous type?

I see this the other way around. If the table tests already define a name for the subtest, there is no need to use the type name, as that was only used to identify the failing test, but that is already referenced through the subtest name.

Reply:
Yep, but custom mappings wouldn't get along nicely with that anonymous type. Anyway, we can take care of it when the time comes 👍